src/main/java/slimeknights/tconstruct/library/utils/RecipeUtil.java

```diff
@@ -0,0 +1,86 @@
+package slimeknights.tconstruct.library.utils;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.NonNullList;
+import net.minecraftforge.oredict.OreDictionary;
+import slimeknights.mantle.client.CreativeTab;
+
+public class RecipeUtil {
+  // list of orePreferences, to be filled by the config
+  private static String[] orePreferences = new String[0];
+  // cache to avoid grabing the same name twice
+  private static Map<String,ItemStack> preferenceCache = new HashMap<>();
+
+  private RecipeUtil() {}
+
+  /**
+   * Called by the config to add the preferences. Do not call outside of Tinkers Construct
+   * @param preferences
+   */
+  public static void setOrePreferences(String[] preferences) {
+    orePreferences = preferences;
+  }
+
+  /**
+   * Gets the preferred ore for the given oredict name
+   * @param oreName  Oredictionary key
+   * @return  The preferred ItemStack, or ItemStack.EMPTY if the name is empty
+   */
+  public static ItemStack getPreference(String oreName) {
+    // if we found one before, use that
+    if(preferenceCache.containsKey(oreName)) {
+      return preferenceCache.get(oreName).copy();
```

Comment:
Very good

Reply:
We used that in some of the model stuff, didn't we. Only downside is we would have to cache if we got no result, if that is acceptable

Reply:
No, the remaining code would have to be adjusted to fit a single call, but that's just some code reordering. You already cache the first oredict entry in your code, it'd just require to move all the remaining code (aka get oredict items) into the cache method.

Reply:
Looking into It, I guess I can just return null from thay function if the item does not exist (so no cache) then use Map.getOrDefault to return ItemStack.EMPTY if the result is null.